src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -34,6 +36,71 @@ class WorkflowDataConditionGroupType(StrEnum):
     WORKFLOW_TRIGGER = "workflow_trigger"
 
 
+def get_actions_to_workflows(actions_to_fire: list[Action]) -> dict[int, Workflow]:
+    action_ids = {action.id for action in actions_to_fire}
+    workflow_actions = (
+        WorkflowDataConditionGroup.objects.select_related("workflow", "condition_group")
```

Comment:
I think that `select_related` is unnecessary, since you're already joining via `values_list`.